From 6890c80750aef97a572f55267ccfe0b61ce2a1f4 Mon Sep 17 00:00:00 2001
From: Sebastian Pipping <sebastian@pipping.org>
Date: Fri, 23 Oct 2026 15:25:54 +0200
Subject: [PATCH] lib: Add missing `NULL` check for
 `parser->m_startDoctypeDeclHandler`

It was possible to reset the handler to `NULL` between the initial check
for `NULL` and the eventual call.
---
 lib/xmlparse.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/lib/xmlparse.c b/lib/xmlparse.c
index e8f1e044..9f03c556 100644
--- a/lib/xmlparse.c
+++ b/lib/xmlparse.c
@@ -5751,7 +5751,7 @@ doProlog(XML_Parser parser, const ENCODING *enc, const char *s, const char *end,
         return XML_ERROR_INVALID_TOKEN;
       }
 
-      if (parser->m_doctypeName) {
+      if (parser->m_doctypeName && parser->m_startDoctypeDeclHandler) {
         beforeHandler(parser);
         parser->m_startDoctypeDeclHandler(
             parser->m_handlerArg, parser->m_doctypeName, parser->m_doctypeSysid,
